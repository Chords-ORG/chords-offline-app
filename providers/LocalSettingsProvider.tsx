import React from "react";
import { LightStyle, DarkStyle } from "../constants/Styles";
import useLocalConfiguration from "../hooks/useLocalConfiguration";
import { useColorScheme } from "react-native";
import { getItem } from "../functions/storage";

export type ChordType = "sharp" | "bemol";
export type Instrument = "guitar" | "piano";
export type CapoConfig = "auto" | "never";
export type LocalColorScheme = "system" | "light" | "dark";

export const CHORD_TYPE_KEY = "chordType";
export const INSTRUMENT_KEY = "instrument";
export const CAPO_CONFIG_KEY = "capoConfig";
export const LOCAL_COLOR_SCHEME_KEY = "localColorScheme";

interface LocalSettingsContextProps {
  chordType: ChordType;
  instrument: Instrument;
  capoConfig: CapoConfig;
  localColorScheme: LocalColorScheme;
  toggleLocalSettings: () => Promise<void>;
}

export const LocalSettingsContext =
  React.createContext<LocalSettingsContextProps>({
    chordType: "sharp",
    instrument: "guitar",
    capoConfig: "auto",
    localColorScheme: "system",
    toggleLocalSettings: async () => {},
  });

interface LocalSettingsProviderProps {
  children: React.ReactNode;
}

export const LocalSettingsProvider = ({
  children,
}: LocalSettingsProviderProps) => {
  const [chordType, setChordTypeState] = React.useState<ChordType>("sharp");
  const [instrument, setInstrumentState] = React.useState<Instrument>("guitar");
  const [capoConfig, setCapoConfigState] = React.useState<CapoConfig>("auto");
  const [localColorScheme, setLocalColorSchemeState] =
    React.useState<LocalColorScheme>("system");

  const getLocalConfiguration = async () => {
    const chordType = await getItem(CHORD_TYPE_KEY);
    const instrument = await getItem(INSTRUMENT_KEY);
    const capoConfig = await getItem(CAPO_CONFIG_KEY);
    const localColorScheme = await getItem(LOCAL_COLOR_SCHEME_KEY);
    return { chordType, instrument, capoConfig, localColorScheme };
  };

  const toggleLocalSettings = React.useCallback(async () => {
    const localConfiguration = await getLocalConfiguration();
    if (localConfiguration.chordType !== null) {
      setChordTypeState(localConfiguration.chordType as ChordType);
    }
    if (localConfiguration.instrument !== null) {
      setInstrumentState(localConfiguration.instrument as Instrument);
    }
    if (localConfiguration.capoConfig !== null) {
      setCapoConfigState(localConfiguration.capoConfig as CapoConfig);
    }
    if (localConfiguration.localColorScheme !== null) {
      setLocalColorSchemeState(
        localConfiguration.localColorScheme as LocalColorScheme
      );
    }
  }, [
    setChordTypeState,
    setInstrumentState,
    setCapoConfigState,
    setLocalColorSchemeState,
  ]);

  React.useEffect(() => {
    toggleLocalSettings();
  }, []);

  return (
    <LocalSettingsContext.Provider
      value={{
        chordType,
        instrument,
        capoConfig,
        localColorScheme,
        toggleLocalSettings,
      }}
    >
      {children}
    </LocalSettingsContext.Provider>
  );
};
