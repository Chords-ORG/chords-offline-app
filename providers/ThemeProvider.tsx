import React from "react";
import { LightStyle, DarkStyle } from "../constants/Styles";
import useLocalConfiguration from "../hooks/useLocalConfiguration";
import { useColorScheme } from "react-native";
import { getItem } from "../functions/storage";

interface ThemeContextProps {
  styleSheet: typeof LightStyle | typeof DarkStyle;
  theme: "light" | "dark";
  toggleTheme: () => void;
}

export const ThemeContext = React.createContext<ThemeContextProps>({
  styleSheet: LightStyle,
  theme: "light",
  toggleTheme: () => {},
});

interface ThemeProviderProps {
  children: React.ReactNode;
}
export const ThemeProvider = ({ children }: ThemeProviderProps) => {
  const [theme, setTheme] = React.useState<"light" | "dark">("light");
  const systemColorScheme = useColorScheme();
  const { localColorScheme } = useLocalConfiguration();

  const toggleTheme = async () => {
    if (localColorScheme === "system") {
      setTheme(systemColorScheme || "light");
    } else {
      setTheme(localColorScheme as "light" | "dark");
    }
  };

  React.useEffect(() => {
    toggleTheme();
  }, [systemColorScheme, localColorScheme]);

  return (
    <ThemeContext.Provider
      value={{
        styleSheet: theme === "light" ? LightStyle : DarkStyle,
        theme,
        toggleTheme,
      }}
    >
      {children}
    </ThemeContext.Provider>
  );
};
