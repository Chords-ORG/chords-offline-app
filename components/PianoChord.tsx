import React, { useState } from "react";
import { StyleSheet, View, Text } from "react-native";
import { TouchableOpacity } from "react-native-gesture-handler";
import { ThemeContext } from "../providers/ThemeProvider";
import { Chord } from "../services/chords";
import PianoChordImage from "./PianoChordImage";

export interface PianoChordProps {
  chordName: string;
}

export default function PianoChord({ chordName }: PianoChordProps) {
  const { colors: themeColors } = React.useContext(ThemeContext);

  const pianoChords: Record<
    string,
    string[][]
  > = require("../assets/data/piano_chords.json");

  const sharpChordName = Chord.toChord(chordName, "sharp");
  const fingersPositionsList = pianoChords[sharpChordName];
  const [idx, setIdx] = useState(0);
  if (!fingersPositionsList || fingersPositionsList.length == 0) {
    return (
      <View
        style={[
          styles.container,
          { justifyContent: "center", alignItems: "center" },
        ]}
      >
        <Text style={[styles.message, { color: themeColors.textPrimary }]}>
          Acorde Indispon√≠vel
        </Text>
      </View>
    );
  } else {
    const tam = fingersPositionsList.length;
    const fingers = fingersPositionsList[idx];
    return (
      <TouchableOpacity onPress={() => setIdx((idx + 1) % tam)}>
        <PianoChordImage fingers={fingers} />
      </TouchableOpacity>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    height: 50,
    width: 200,
  },
  message: {
    fontFamily: "roboto-bold",
    fontSize: 12,
    textAlign: "center",
  },
});
