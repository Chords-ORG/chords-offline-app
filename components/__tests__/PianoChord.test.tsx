import renderer from "react-test-renderer";
import React from "react";
import { ThemeContext } from "../../providers/ThemeProvider";
import {
  DarkColors,
  DarkStyle,
  LightColors,
  LightStyle,
} from "../../constants/Styles";
import PianoChord from "../PianoChord";
import { TouchableOpacity } from "react-native-gesture-handler";
import { act } from "react-dom/test-utils";

jest.mock("@react-native-async-storage/async-storage", () =>
  require("@react-native-async-storage/async-storage/jest/async-storage-mock")
);

describe("<PianoChord />", () => {
  describe("Testing rendering", () => {
    it(`should render C on light theme`, () => {
      const tree = renderer
        .create(
          <ThemeContext.Provider
            value={{
              styleSheet: LightStyle,
              colors: LightColors,
              theme: "light",
            }}
          >
            <PianoChord chordName="C" />
          </ThemeContext.Provider>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it(`should render C on dark theme`, () => {
      const tree = renderer

        .create(
          <ThemeContext.Provider
            value={{
              styleSheet: DarkStyle,
              colors: DarkColors,
              theme: "dark",
            }}
          >
            <PianoChord chordName="C" />
          </ThemeContext.Provider>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe("Testing clicking and changing note", () => {
    const tree = renderer.create(
      <ThemeContext.Provider
        value={{
          styleSheet: LightStyle,
          colors: LightColors,
          theme: "light",
        }}
      >
        <PianoChord chordName="C" />
      </ThemeContext.Provider>
    );

    it("should render correcly after changing the chord form", () => {
      expect(tree.toJSON()).toMatchSnapshot();
      for (let i = 0; i < 10; i++) {
        act(() => {
          tree.root.findByType(TouchableOpacity).props.onPress();
        });
        expect(tree.toJSON()).toMatchSnapshot();
      }
    });
  });

  describe("Testing chord not found", () => {
    it("should render correctly when the chord is not found", () => {
      const tree = renderer
        .create(
          <ThemeContext.Provider
            value={{
              styleSheet: LightStyle,
              colors: LightColors,
              theme: "light",
            }}
          >
            <PianoChord chordName="a#dajsklkj" />
          </ThemeContext.Provider>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
