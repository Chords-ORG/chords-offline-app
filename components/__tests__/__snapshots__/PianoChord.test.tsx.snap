// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PianoChord /> Testing chord not found should render correctly when the chord is not found 1`] = `
<View
  style={
    [
      {
        "height": 50,
        "width": 200,
      },
      {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "fontFamily": "roboto-bold",
          "fontSize": 12,
          "textAlign": "center",
        },
        {
          "color": "#333333",
        },
      ]
    }
  >
     
    Acorde Indispon√≠vel
     
  </Text>
</View>
`;

exports[`<PianoChord /> Testing clicking and changing note should render correcly after changing the chord form 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  delayLongPress={600}
  enabled={true}
  exclusive={true}
  handlerTag={1}
  handlerType="NativeViewGestureHandler"
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  touchSoundDisabled={false}
>
  <View
    accessible={true}
    collapsable={false}
    style={
      {
        "height": 50,
        "opacity": 1,
        "width": 200,
      }
    }
  >
    <Image
      source={1}
      style={
        {
          "height": 50,
          "width": 200,
        }
      }
    />
    <Image
      source={1}
      style={
        {
          "bottom": 2,
          "height": 8,
          "left": 2,
          "position": "absolute",
          "width": 8,
        }
      }
    />
    <Image
      source={1}
      style={
        {
          "bottom": 2,
          "height": 8,
          "left": 26,
          "position": "absolute",
          "width": 8,
        }
      }
    />
    <Image
      source={1}
      style={
        {
          "bottom": 2,
          "height": 8,
          "left": 49,
          "position": "absolute",
          "width": 8,
        }
      }
    />
  </View>
</RNGestureHandlerButton>
`;

exports[`<PianoChord /> Testing clicking and changing note should render correcly after changing the chord form 2`] = `
<RNGestureHandlerButton
  collapsable={false}
  delayLongPress={600}
  enabled={true}
  exclusive={true}
  handlerTag={1}
  handlerType="NativeViewGestureHandler"
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  touchSoundDisabled={false}
>
  <View
    accessible={true}
    collapsable={false}
    style={
      {
        "height": 50,
        "opacity": 1,
        "width": 200,
      }
    }
  >
    <Image
      source={1}
      style={
        {
          "height": 50,
          "width": 200,
        }
      }
    />
    <Image
      source={1}
      style={
        {
          "bottom": 2,
          "height": 8,
          "left": 2,
          "position": "absolute",
          "width": 8,
        }
      }
    />
    <Image
      source={1}
      style={
        {
          "bottom": 2,
          "height": 8,
          "left": 26,
          "position": "absolute",
          "width": 8,
        }
      }
    />
    <Image
      source={1}
      style={
        {
          "bottom": 2,
          "height": 8,
          "left": 49,
          "position": "absolute",
          "width": 8,
        }
      }
    />
  </View>
</RNGestureHandlerButton>
`;

exports[`<PianoChord /> Testing clicking and changing note should render correcly after changing the chord form 3`] = `
<RNGestureHandlerButton
  collapsable={false}
  delayLongPress={600}
  enabled={true}
  exclusive={true}
  handlerTag={1}
  handlerType="NativeViewGestureHandler"
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  touchSoundDisabled={false}
>
  <View
    accessible={true}
    collapsable={false}
    style={
      {
        "height": 50,
        "opacity": 1,
        "width": 200,
      }
    }
  >
    <Image
      source={1}
      style={
        {
          "height": 50,
          "width": 200,
        }
      }
    />
    <Image
      source={1}
      style={
        {
          "bottom": 2,
          "height": 8,
          "left": 2,
          "position": "absolute",
          "width": 8,
        }
      }
    />
    <Image
      source={1}
      style={
        {
          "bottom": 2,
          "height": 8,
          "left": 26,
          "position": "absolute",
          "width": 8,
        }
      }
    />
    <Image
      source={1}
      style={
        {
          "bottom": 2,
          "height": 8,
          "left": 49,
          "position": "absolute",
          "width": 8,
        }
      }
    />
  </View>
</RNGestureHandlerButton>
`;

exports[`<PianoChord /> Testing clicking and changing note should render correcly after changing the chord form 4`] = `
<RNGestureHandlerButton
  collapsable={false}
  delayLongPress={600}
  enabled={true}
  exclusive={true}
  handlerTag={1}
  handlerType="NativeViewGestureHandler"
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  touchSoundDisabled={false}
>
  <View
    accessible={true}
    collapsable={false}
    style={
      {
        "height": 50,
        "opacity": 1,
        "width": 200,
      }
    }
  >
    <Image
      source={1}
      style={
        {
          "height": 50,
          "width": 200,
        }
      }
    />
    <Image
      source={1}
      style={
        {
          "bottom": 2,
          "height": 8,
          "left": 2,
          "position": "absolute",
          "width": 8,
        }
      }
    />
    <Image
      source={1}
      style={
        {
          "bottom": 2,
          "height": 8,
          "left": 26,
          "position": "absolute",
          "width": 8,
        }
      }
    />
    <Image
      source={1}
      style={
        {
          "bottom": 2,
          "height": 8,
          "left": 49,
          "position": "absolute",
          "width": 8,
        }
      }
    />
  </View>
</RNGestureHandlerButton>
`;

exports[`<PianoChord /> Testing clicking and changing note should render correcly after changing the chord form 5`] = `
<RNGestureHandlerButton
  collapsable={false}
  delayLongPress={600}
  enabled={true}
  exclusive={true}
  handlerTag={1}
  handlerType="NativeViewGestureHandler"
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  touchSoundDisabled={false}
>
  <View
    accessible={true}
    collapsable={false}
    style={
      {
        "height": 50,
        "opacity": 1,
        "width": 200,
      }
    }
  >
    <Image
      source={1}
      style={
        {
          "height": 50,
          "width": 200,
        }
      }
    />
    <Image
      source={1}
      style={
        {
          "bottom": 2,
          "height": 8,
          "left": 2,
          "position": "absolute",
          "width": 8,
        }
      }
    />
    <Image
      source={1}
      style={
        {
          "bottom": 2,
          "height": 8,
          "left": 26,
          "position": "absolute",
          "width": 8,
        }
      }
    />
    <Image
      source={1}
      style={
        {
          "bottom": 2,
          "height": 8,
          "left": 49,
          "position": "absolute",
          "width": 8,
        }
      }
    />
  </View>
</RNGestureHandlerButton>
`;

exports[`<PianoChord /> Testing clicking and changing note should render correcly after changing the chord form 6`] = `
<RNGestureHandlerButton
  collapsable={false}
  delayLongPress={600}
  enabled={true}
  exclusive={true}
  handlerTag={1}
  handlerType="NativeViewGestureHandler"
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  touchSoundDisabled={false}
>
  <View
    accessible={true}
    collapsable={false}
    style={
      {
        "height": 50,
        "opacity": 1,
        "width": 200,
      }
    }
  >
    <Image
      source={1}
      style={
        {
          "height": 50,
          "width": 200,
        }
      }
    />
    <Image
      source={1}
      style={
        {
          "bottom": 2,
          "height": 8,
          "left": 2,
          "position": "absolute",
          "width": 8,
        }
      }
    />
    <Image
      source={1}
      style={
        {
          "bottom": 2,
          "height": 8,
          "left": 26,
          "position": "absolute",
          "width": 8,
        }
      }
    />
    <Image
      source={1}
      style={
        {
          "bottom": 2,
          "height": 8,
          "left": 49,
          "position": "absolute",
          "width": 8,
        }
      }
    />
  </View>
</RNGestureHandlerButton>
`;

exports[`<PianoChord /> Testing clicking and changing note should render correcly after changing the chord form 7`] = `
<RNGestureHandlerButton
  collapsable={false}
  delayLongPress={600}
  enabled={true}
  exclusive={true}
  handlerTag={1}
  handlerType="NativeViewGestureHandler"
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  touchSoundDisabled={false}
>
  <View
    accessible={true}
    collapsable={false}
    style={
      {
        "height": 50,
        "opacity": 1,
        "width": 200,
      }
    }
  >
    <Image
      source={1}
      style={
        {
          "height": 50,
          "width": 200,
        }
      }
    />
    <Image
      source={1}
      style={
        {
          "bottom": 2,
          "height": 8,
          "left": 2,
          "position": "absolute",
          "width": 8,
        }
      }
    />
    <Image
      source={1}
      style={
        {
          "bottom": 2,
          "height": 8,
          "left": 26,
          "position": "absolute",
          "width": 8,
        }
      }
    />
    <Image
      source={1}
      style={
        {
          "bottom": 2,
          "height": 8,
          "left": 49,
          "position": "absolute",
          "width": 8,
        }
      }
    />
  </View>
</RNGestureHandlerButton>
`;

exports[`<PianoChord /> Testing clicking and changing note should render correcly after changing the chord form 8`] = `
<RNGestureHandlerButton
  collapsable={false}
  delayLongPress={600}
  enabled={true}
  exclusive={true}
  handlerTag={1}
  handlerType="NativeViewGestureHandler"
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  touchSoundDisabled={false}
>
  <View
    accessible={true}
    collapsable={false}
    style={
      {
        "height": 50,
        "opacity": 1,
        "width": 200,
      }
    }
  >
    <Image
      source={1}
      style={
        {
          "height": 50,
          "width": 200,
        }
      }
    />
    <Image
      source={1}
      style={
        {
          "bottom": 2,
          "height": 8,
          "left": 2,
          "position": "absolute",
          "width": 8,
        }
      }
    />
    <Image
      source={1}
      style={
        {
          "bottom": 2,
          "height": 8,
          "left": 26,
          "position": "absolute",
          "width": 8,
        }
      }
    />
    <Image
      source={1}
      style={
        {
          "bottom": 2,
          "height": 8,
          "left": 49,
          "position": "absolute",
          "width": 8,
        }
      }
    />
  </View>
</RNGestureHandlerButton>
`;

exports[`<PianoChord /> Testing clicking and changing note should render correcly after changing the chord form 9`] = `
<RNGestureHandlerButton
  collapsable={false}
  delayLongPress={600}
  enabled={true}
  exclusive={true}
  handlerTag={1}
  handlerType="NativeViewGestureHandler"
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  touchSoundDisabled={false}
>
  <View
    accessible={true}
    collapsable={false}
    style={
      {
        "height": 50,
        "opacity": 1,
        "width": 200,
      }
    }
  >
    <Image
      source={1}
      style={
        {
          "height": 50,
          "width": 200,
        }
      }
    />
    <Image
      source={1}
      style={
        {
          "bottom": 2,
          "height": 8,
          "left": 2,
          "position": "absolute",
          "width": 8,
        }
      }
    />
    <Image
      source={1}
      style={
        {
          "bottom": 2,
          "height": 8,
          "left": 26,
          "position": "absolute",
          "width": 8,
        }
      }
    />
    <Image
      source={1}
      style={
        {
          "bottom": 2,
          "height": 8,
          "left": 49,
          "position": "absolute",
          "width": 8,
        }
      }
    />
  </View>
</RNGestureHandlerButton>
`;

exports[`<PianoChord /> Testing clicking and changing note should render correcly after changing the chord form 10`] = `
<RNGestureHandlerButton
  collapsable={false}
  delayLongPress={600}
  enabled={true}
  exclusive={true}
  handlerTag={1}
  handlerType="NativeViewGestureHandler"
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  touchSoundDisabled={false}
>
  <View
    accessible={true}
    collapsable={false}
    style={
      {
        "height": 50,
        "opacity": 1,
        "width": 200,
      }
    }
  >
    <Image
      source={1}
      style={
        {
          "height": 50,
          "width": 200,
        }
      }
    />
    <Image
      source={1}
      style={
        {
          "bottom": 2,
          "height": 8,
          "left": 2,
          "position": "absolute",
          "width": 8,
        }
      }
    />
    <Image
      source={1}
      style={
        {
          "bottom": 2,
          "height": 8,
          "left": 26,
          "position": "absolute",
          "width": 8,
        }
      }
    />
    <Image
      source={1}
      style={
        {
          "bottom": 2,
          "height": 8,
          "left": 49,
          "position": "absolute",
          "width": 8,
        }
      }
    />
  </View>
</RNGestureHandlerButton>
`;

exports[`<PianoChord /> Testing clicking and changing note should render correcly after changing the chord form 11`] = `
<RNGestureHandlerButton
  collapsable={false}
  delayLongPress={600}
  enabled={true}
  exclusive={true}
  handlerTag={1}
  handlerType="NativeViewGestureHandler"
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  touchSoundDisabled={false}
>
  <View
    accessible={true}
    collapsable={false}
    style={
      {
        "height": 50,
        "opacity": 1,
        "width": 200,
      }
    }
  >
    <Image
      source={1}
      style={
        {
          "height": 50,
          "width": 200,
        }
      }
    />
    <Image
      source={1}
      style={
        {
          "bottom": 2,
          "height": 8,
          "left": 2,
          "position": "absolute",
          "width": 8,
        }
      }
    />
    <Image
      source={1}
      style={
        {
          "bottom": 2,
          "height": 8,
          "left": 26,
          "position": "absolute",
          "width": 8,
        }
      }
    />
    <Image
      source={1}
      style={
        {
          "bottom": 2,
          "height": 8,
          "left": 49,
          "position": "absolute",
          "width": 8,
        }
      }
    />
  </View>
</RNGestureHandlerButton>
`;

exports[`<PianoChord /> Testing rendering should render C on dark theme 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  delayLongPress={600}
  enabled={true}
  exclusive={true}
  handlerTag={3}
  handlerType="NativeViewGestureHandler"
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  touchSoundDisabled={false}
>
  <View
    accessible={true}
    collapsable={false}
    style={
      {
        "height": 50,
        "opacity": 1,
        "width": 200,
      }
    }
  >
    <Image
      source={1}
      style={
        {
          "height": 50,
          "width": 200,
        }
      }
    />
    <Image
      source={1}
      style={
        {
          "bottom": 2,
          "height": 8,
          "left": 2,
          "position": "absolute",
          "width": 8,
        }
      }
    />
    <Image
      source={1}
      style={
        {
          "bottom": 2,
          "height": 8,
          "left": 26,
          "position": "absolute",
          "width": 8,
        }
      }
    />
    <Image
      source={1}
      style={
        {
          "bottom": 2,
          "height": 8,
          "left": 49,
          "position": "absolute",
          "width": 8,
        }
      }
    />
  </View>
</RNGestureHandlerButton>
`;

exports[`<PianoChord /> Testing rendering should render C on light theme 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  delayLongPress={600}
  enabled={true}
  exclusive={true}
  handlerTag={2}
  handlerType="NativeViewGestureHandler"
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  touchSoundDisabled={false}
>
  <View
    accessible={true}
    collapsable={false}
    style={
      {
        "height": 50,
        "opacity": 1,
        "width": 200,
      }
    }
  >
    <Image
      source={1}
      style={
        {
          "height": 50,
          "width": 200,
        }
      }
    />
    <Image
      source={1}
      style={
        {
          "bottom": 2,
          "height": 8,
          "left": 2,
          "position": "absolute",
          "width": 8,
        }
      }
    />
    <Image
      source={1}
      style={
        {
          "bottom": 2,
          "height": 8,
          "left": 26,
          "position": "absolute",
          "width": 8,
        }
      }
    />
    <Image
      source={1}
      style={
        {
          "bottom": 2,
          "height": 8,
          "left": 49,
          "position": "absolute",
          "width": 8,
        }
      }
    />
  </View>
</RNGestureHandlerButton>
`;
